syntax = "proto3";

message MsgRequest {
	int32 id = 1;
	string type = 2;
	string interface = 3;
	int32 money = 4;
	int32 clock = 5;
}

message MsgResponse {
	int32 id = 1;
	string interface = 2;
	string result = 3;
	int32 money = 4;
	int32 clock = 5;
}

service Branch {
	rpc MsgDelivery(MsgRequest) returns (MsgResponse) {}
	rpc MsgPropagation(MsgRequest) returns (MsgResponse) {}
}